python/polycap.pyx

```diff
@@ -544,22 +1023,104 @@ cdef class Photon:
 
         return weights_np
 
+    @property
+    def extleaks(self):
+        '''Retrieve exterior :ref:``Leak`` class array from a :ref:``Photon`` class '''
+        if self._photon is NULL:
+            return None
+
+        cdef polycap_error *error = NULL
+
+        if self._n_ext_leaks == 0:
+            polycap_photon_get_extleak_data(self._photon, &self._ext_leaks, &self._n_ext_leaks, &error)
+            polycap_set_exception(error)
+
+        if self._n_ext_leaks > 0:
+            for i in range(self._n_ext_leaks):
+                leak = Leak.create(self._ext_leaks[i])
+                yield leak
+        else:
+            return None
+
+    @property
+    def intleaks(self):
+        '''Retrieve interior :ref:``Leak`` class array from a :ref:``Photon`` class '''
+        if self._photon is NULL:
+            return None
+
+        cdef polycap_error *error = NULL
+
+        #logger.debug('Before calling C')
+        if self._n_int_leaks == 0:
+            polycap_photon_get_intleak_data(self._photon, &self._int_leaks, &self._n_int_leaks, &error)
+            polycap_set_exception(error)
+        #logger.debug('After calling C') 
+
+        rv = list()
+
+        if self._n_int_leaks > 0:
+            for i in range(self._n_int_leaks):
+                leak = Leak.create(self._int_leaks[i])
+                yield leak
+        else:
+            return None
+
     def get_exit_coords(self):
         '''Retrieve exit coordinates from a :ref:``Photon`` class'''
-        return vector2tuple(polycap_photon_get_exit_coords(self.photon))
+        return vector2tuple(polycap_photon_get_exit_coords(self._photon))
+
+    @property
+    def exit_coords(self):
+        '''Retrieve exit coordinates from a :ref:``Photon`` class'''
+        return vector2tuple(polycap_photon_get_exit_coords(self._photon))
+
+    @property
+    def start_coords(self):
+        '''Retrieve start coordinates from a :ref:``Photon`` class'''
+        return vector2tuple(polycap_photon_get_start_coords(self._photon))
 
     def get_exit_direction(self):
         '''Retrieve exit direction from a :ref:``Photon`` class'''
-        return vector2tuple(polycap_photon_get_exit_direction(self.photon))
+        return vector2tuple(polycap_photon_get_exit_direction(self._photon))
+
+    @property
+    def exit_direction(self):
+        '''Retrieve exit direction from a :ref:``Photon`` class'''
+        return vector2tuple(polycap_photon_get_exit_direction(self._photon))
+
+    @property
+    def start_direction(self):
+        '''Retrieve start direction from a :ref:``Photon`` class'''
+        return vector2tuple(polycap_photon_get_start_direction(self._photon))
 
     def get_exit_electric_vector(self):
```

Comment:
Het is best dat je de `get_` methods herschrijft zodat ze de corresponderende property gebruiken:

```python
def get_exit_electric_vector(self):
    return self.exit_electric_vector
```